packages/fx-core/src/component/driver/aad/utility/aadAppClient.ts

```diff
@@ -114,18 +116,32 @@ export class AadAppClient {
       },
     };
 
-    const response = await this.axios.post(`applications/${objectId}/addPassword`, requestBody, {
-      "axios-retry": {
-        retries: this.retryNumber,
-        retryDelay: axiosRetry.exponentialDelay,
-        retryCondition: (error) =>
-          axiosRetry.isNetworkError(error) ||
-          axiosRetry.isRetryableError(error) ||
-          this.is404Error(error), // also retry 404 error since Microsoft Entra need sometime to sync created Microsoft Entra app data
-      },
-    });
+    try {
+      const response = await this.axios.post(`applications/${objectId}/addPassword`, requestBody, {
+        "axios-retry": {
+          retries: this.retryNumber,
+          retryDelay: axiosRetry.exponentialDelay,
+          retryCondition: (error) =>
+            axiosRetry.isNetworkError(error) ||
+            axiosRetry.isRetryableError(error) ||
+            this.is404Error(error), // also retry 404 error since Microsoft Entra need sometime to sync created Microsoft Entra app data
+        },
+      });
 
-    return response.data.secretText;
+      return response.data.secretText;
```

Comment:
is `response.data.secretText!` better?

Reply:
As synced offline, since data is Record<any,any> and bypasses the type check, we don't need to add `!` here.